src/main/kotlin/net/atos/zac/app/informatieobjecten/InformatieObjectenRESTService.kt

```diff
@@ -0,0 +1,675 @@
+    ): RESTEnkelvoudigInformatieobject =
+        uuid
+            .let(drcClientService::readEnkelvoudigInformatieobject)
+            .let { enkelvoudigInformatieObject ->
+                zaakUUID?.let(zrcClientService::readZaak)?.let {
+                    informatieobjectConverter.convertToREST(enkelvoudigInformatieObject, it)
+                } ?: informatieobjectConverter.convertToREST(enkelvoudigInformatieObject)
+            }
+
+    @GET
+    @Path("informatieobject/versie/{uuid}/{version}")
+    fun readEnkelvoudigInformatieobject(
+        @PathParam("uuid") uuid: UUID,
+        @PathParam("version") version: Int
+    ): RESTEnkelvoudigInformatieobject = uuid
+        .let(drcClientService::readEnkelvoudigInformatieobject)
+        .let { currentVersion ->
+            return when {
+                version < currentVersion.versie -> informatieobjectConverter.convertToREST(
+                    drcClientService.readEnkelvoudigInformatieobjectVersie(uuid, version)
+                )
+                else -> informatieobjectConverter.convertToREST(currentVersion)
+            }
+        }
+
+    @PUT
+    @Path("informatieobjectenList")
+    fun listEnkelvoudigInformatieobjecten(
+        zoekParameters: RESTInformatieobjectZoekParameters
+    ): List<RESTEnkelvoudigInformatieobject> {
+        val zaak = zoekParameters.zaakUUID?.let { zrcClientService.readZaak(it) }
+        zoekParameters.informatieobjectUUIDs?.let {
+            return informatieobjectConverter.convertUUIDsToREST(it, zaak)
+        } ?: run {
+            checkNotNull(zaak) { "Zoekparameters hebben geen waarde" }
+            assertPolicy(policyService.readZaakRechten(zaak).lezen)
+            var enkelvoudigInformatieobjectenVoorZaak = listEnkelvoudigInformatieobjectenVoorZaak(zaak)
+            if (zoekParameters.gekoppeldeZaakDocumenten) {
+                enkelvoudigInformatieobjectenVoorZaak.addAll(listGekoppeldeZaakInformatieObjectenVoorZaak(zaak))
+            }
+            zoekParameters.besluittypeUUID?.let {
+                val besluittype = ztcClientService.readBesluittype(zoekParameters.besluittypeUUID)
+                val compareList = besluittype.informatieobjecttypen.stream().map { UriUtil.uuidFromURI(it) }.toList()
+                enkelvoudigInformatieobjectenVoorZaak = enkelvoudigInformatieobjectenVoorZaak.filter {
+                    compareList.contains(it.informatieobjectTypeUUID)
+                }.toMutableList()
+            }
+            return enkelvoudigInformatieobjectenVoorZaak
+        }
+    }
+
+    @GET
+    @Path("informatieobjecten/zaak/{zaakUuid}/teVerzenden")
+    fun listEnkelvoudigInformatieobjectenVoorVerzenden(
+        @PathParam("zaakUuid") zaakUuid: UUID
+    ): List<RESTEnkelvoudigInformatieobject> {
+        val zaak = zrcClientService.readZaak(zaakUuid)
+        assertPolicy(policyService.readZaakRechten(zaak).lezen)
+        return zrcClientService.listZaakinformatieobjecten(zaak).stream()
```

Comment:
```suggestion
        return zrcClientService.listZaakinformatieobjecten(zaak)
```

? Java's `stream` is usually not needed in Kotlin

Reply:
we have `findAny()` in the call chain that seems a harder to replace